core/node/rpc/add_event.go

```diff
@@ -78,9 +78,54 @@ func (s *Service) addParsedEvent(
 	}
 
 	if chainAuthArgs != nil {
-		err := s.chainAuth.IsEntitled(ctx, s.config, chainAuthArgs)
-		if err != nil {
-			return err
+		chainAuthErr := s.chainAuth.IsEntitled(ctx, s.config, chainAuthArgs)
+		if chainAuthErr != nil {
```

Comment:
how do we know this error is an actual permission denied and not a timeout or network failure?